semgrep/semgrep/error.py

```diff
@@ -94,19 +111,21 @@ def __init__(
         self.level = level
         self.spans = spans
         self.help = help
-
-    @property
-    def code(self) -> int:  # type: ignore
-        """
-        If we explicitly set a code, return that.
-        Otherwise, if the cause sets a code, return that.
-        Otherwise, return a generic FATAL_EXIT_CODE
-        """
-        return (
-            self.__dict__.get("code")
-            or getattr(self.__cause__, "code", None)
-            or FATAL_EXIT_CODE
+        assert hasattr(
```

Comment:
Is this the best way to enforce this?

Reply:
I suppose we could write a semgrep rule :-)